.travis.yml

```diff
@@ -24,13 +24,15 @@ env:
   - REMOVE_XDEBUG="0"
   - REMOVE_XDEBUG="1"
 
-sudo: false
+sudo: required
 
 cache:
   directories:
     - $HOME/.composer/cache
 
 before_install:
+  - sudo apt-get update --yes
+  - sudo apt-get install --yes net-tools
```

Comment:
fwiw, not sure if/how you can do this in PHP, but in Python you can do this by binding a socket to port 0 (which is standard Linux stuff), and port 0 means "just give me a port, any port". Then You can inspect the socket to get what port it actually bound to.

Something like:

```python
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
s.bind((host, 0))
port = s.getsockname()[1]
s.close()
```

Not that it particularly matters, just sharing a tip if you wanted to avoid trying to use `netstat` to discover an available port.